src/plugins/vis_builder/public/embeddable/vis_builder_embeddable.tsx

```diff
@@ -28,28 +28,36 @@ import {
   TimeRange,
 } from '../../../data/public';
 import { validateSchemaState } from '../application/utils/validations/validate_schema_state';
-import { getExpressionLoader, getTypeService } from '../plugin_services';
+import {
+  getExpressionLoader,
+  getIndexPatterns,
+  getTypeService,
+  getUIActions,
+} from '../plugin_services';
 import { PersistedState } from '../../../visualizations/public';
-import { RenderState, VisualizationState } from '../application/utils/state_management';
+import { VisBuilderSavedVis } from '../saved_visualizations/transforms';
+import { handleVisEvent } from '../application/utils/handle_vis_event';
 
 // Apparently this needs to match the saved object type for the clone and replace panel actions to work
 export const VISBUILDER_EMBEDDABLE = VISBUILDER_SAVED_OBJECT;
 
 export interface VisBuilderEmbeddableConfiguration {
-  savedVisBuilder: VisBuilderSavedObjectAttributes;
+  savedVis: VisBuilderSavedVis;
   // TODO: add indexPatterns as part of configuration
   // indexPatterns?: IIndexPattern[];
   editPath: string;
   editUrl: string;
   editable: boolean;
 }
 
+export type VisBuilderInput = SavedObjectEmbeddableInput;
```

Comment:
You are right, that is much more explicit. However, since the UI state PR also affects the same files, i'd avoid making this change now though. Is that okay?